### PR TITLE
cli: add "scan" subcommand for standalone hash reports

### LINKED ISSUE #17: cli: new subcommand "scan" (standalone hash/scan report)
## Objective

Expose `FileScanner` standalone: `datrom scan <dirs...> [--out-file <path>] [--out-mode xml|json|yaml]` producing a hash/scan report without requiring a DAT or the 1G1R pipeline.

## Details

Useful on its own (inventory, hash verification) and as the future GUI's scan-preview backend. Reuses `PerformanceOptions` scanner knobs and the existing progress bar listeners.

## Required reading

- `core/.../io/FileScanner.java` (scan results shape)
- `cli/.../option/PerformanceOptions.java`, `cli/.../progressbar/`
- `core/.../SerializationHelper.java` for report serialization

## Constraints

- Report schema is the serialized core scan result — no bespoke report DTO in `cli`.

## Verification

- Scanning a `test-data/` fixture directory yields expected entries and hashes (existing scanner test fixtures reused where possible).
- Output modes produce parseable JSON/YAML/XML.
- `mvn verify` green.

## Escalation

If scan results are not cleanly serializable as-is, post the model gap here before adding annotations to core types.

### PR DESCRIPTION
**Pull request by andrebrait**
Fixes #17

## What

`datrom scan <dirs...> [--out-mode xml|json|yaml] [--out-file <path>] [scan performance options]` — exposes `FileScanner` standalone: hash/inventory report over directories and archives, no DAT required (empty datafiles/detectors fall back to `FileScannerParameters.withDefaults()`, i.e. no header detection). The report serializes the core scan results directly — no bespoke DTO; results are copied to a plain list only to keep the XML root tag sane (`<ArrayList>`, not Guava's package-private type).

stdout stays machine-readable: the jline terminal is built with `SystemOutput.ForcedSysErr`, so progress bars go to stderr unconditionally (jar-probed: 0 ANSI escapes on stdout, progress frames on stderr, output parses as JSON/YAML/XML). Exits 1 when any file failed to scan, 0 otherwise — deliberate divergence from `1g1r`'s always-0 (a report tool should signal partial failure).

## Evidence

- Red→green surface proof: `ScanMigrationTest` RED on master (`Unmatched arguments... 'scan'`), GREEN after; re-executed by the verifier gate via `verify-red-proof.sh`.
- Known-hash fixture test verified non-circular by the gate: expected CRC/MD5/SHA1 match `test-data`'s checksum manifests and an independent `shasum` run.
- Gate mutation (JSON branch writing YAML) fails 2 tests; stdout-cleanliness test exercises the real terminal/progress path (`CommandLineProgressBar.init()` writes ANSI unconditionally).
- Full `mvn verify` green; no core/domain changes.

## Note

`scan --help` shows the copy-side performance flags (`--copy-threads` etc.) because the shared `PerformanceOptions` ArgGroup bundles scan+copy knobs — pre-existing bundling, harmless (unused flags are ignored); splitting the group is a possible future polish item.

🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

## Summary by CodeRabbit

* **New Features**
  * Added a `scan` command for scanning directories for ROMs and archives.
  * Supports JSON, YAML, and XML output, with optional output-file saving.
  * Added configurable scanning performance options and exit status reporting.

* **Bug Fixes**
  * Improved progress display stability for empty scans, limited files, and inactive threads.

* **Tests**
  * Added coverage for scan output formats, validation, configuration options, and edge cases.

### CODE REVIEW

**Comment by andrebrait** [comment]
## Review audit

**CodeRabbit**: quota-limited (~13 min) — dropped per the 5-minute rule. Two independent passes ran instead.

**Adversarial review #1**: model `claude-sonnet-5`, profile `full`, effort `xhigh` (fresh angles on top of the internal verifier gate).
**Second pass**: model `claude-fable-5` (top tier), line-level correctness read.

Findings ledger:

1. BLOCKING (review #1) — empty-directory scan crashed with `ArithmeticException: / by zero` in `CommandLineProgressBar.printMainBar` (jar-executed, all three out-modes). **Fixed @ `d5f2f62`** (guard; red-first test `emptyDirectoryScanProducesEmptyReport`, frozen `e733830a`).
2. BLOCKING (review #1) — scanning fewer files than the configured thread count crashed with an NPE in `getFinalAverage` (jar-executed, 2 files vs default 5 threads). **Fixed @ `d5f2f62`** (null-slot skip; red-first test `scanningFewerFilesThanThreadsSucceeds`, same frozen file). The fix commit also guards the `log10(0)` field-width path and enumerates every other division/indexing shape in the class as safe.
3. MAJOR test-honesty (review #2 F1) — the stdout-cleanliness test is blind to a `ForcedSysErr` regression because jline writes at file-descriptor level. **Fixed @ `be8b79b`**: `PROGRESS_OUTPUT` constant is now the testable surface, pinned by `progressOutputStaysPinnedToStderr`; the end-to-end property remains jar-probe-verified.
4. MINOR (review #2 F2) — listing-stage failures and swallowed futures never set the scanner error flag → exit 0 on an incomplete report. Pre-existing core behavior; **deferred: #37**.
5. MINOR (review #2 F3) — `--out-file` requires a pre-existing parent dir and writes non-atomically. Consistent with existing precedent; **skipped as polish** (noted here).
6. INFO — once-in-~40-runs stray `#` in captured test stdout; fix round investigated (time-boxed): no deterministic cause, 0/40 repro, both plausible leak paths ruled out; documented in the `d5f2f62` commit body — will file an issue if it recurs with raw-byte capture.
7. Review #1 confirmations: report entry order deterministic (sorted submission + sequential future joins, byte-identical across repeated runs incl. tie fixtures); unreadable-file path exits 1 with a partial report once the crash fix landed; no field narrowing vs what the 1G1R matcher consumes.

**Delta re-review** (`claude-sonnet-5`, base `ec061d1`, scope `d5f2f62`+`be8b79b`): PASS — freeze hash verified against the `d5f2f62` blob, both crashes re-reproduced RED at base and GREEN at HEAD, jar re-probed (empty dir → `[]` exit 0; 2 files/4 threads → 2 entries exit 0), guards confirmed normal-path-neutral, pin confirmed live-wired.

**CI**: gated on the current head run.

🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

File: cli/src/main/java/io/github/datromtool/cli/command/ScanCommand.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    @Override
+    public Integer call() {
+        AppConfig.FileScannerConfig scannerConfig = resolveScannerConfig();
+        FileScannerLoggingListener loggingListener = new FileScannerLoggingListener();
+        ImmutableList<FileScanner.Result> scanResults = ImmutableList.of();
+        try (Terminal terminal = createTerminal()) {
+            List<FileScanner.Listener> listeners = ImmutableList.of(
+                    loggingListener,
+                    new CommandLineProgressBar(terminal, "Scanning", "Scanning input directories..."));
```
[the remark this answers is not in the document]
Comment (andrebrait) [reply cli/src/main/java/io/github/datromtool/cli/command/ScanCommand.java:106]: Deferred to #38 (review arrived after the merge). Confirmed real hardening: filenames with embedded control characters render unfiltered through the progress-bar labels; the issue carries the gate record and reproduction.

🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

File: cli/src/main/java/io/github/datromtool/cli/command/ScanCommand.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                Files.write(outFile, lines);
+            } catch (IOException e) {
+                throw new CommandLine.ParameterException(
+                        commandSpec.commandLine(),
+                        format("Could not write output file '%s': %s", outFile, e.getMessage()));
+            }
+        } else {
+            lines.forEach(System.out::println);
+        }
+        return loggingListener.isErrors() ? 1 : 0;
```
[the remark this answers is not in the document]
Comment (andrebrait) [reply cli/src/main/java/io/github/datromtool/cli/command/ScanCommand.java:142]: Deferred to #37 (filed during this PR's review round, before your comment arrived): FileScanner's listing-stage failures and swallowed futures never reach the listener/error contract, so exit 0 after a partial scan — the fix belongs in core's scanner error propagation, which that issue owns.

🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.
